spec/entities/base_spec.rb

```diff
@@ -113,103 +187,181 @@ class Child < KalibroClient::Entities::Base; end
-          expect(subject.kalibro_errors[0]).to eq("Name has already been taken")
+        it 'is expected to call the update method'  do
+          subject.expects(:update).returns(true)
+          expect(subject.save).to eq(true)
         end
       end
     end
+  end
+
+  describe 'update' do
+    it_behaves_like 'persistence method', :update, :put
 
-    context 'when it is persisted' do
+    context 'with valid parameters' do
       before :each do
-        subject.persisted = true
-      end
+        id = 42
 
-      it 'is expected to call the update method'  do
-        subject.expects(:update)
+        subject.stubs(:id).returns(id)
+        KalibroClient::Entities::Base.expects(:request).with(':id', has_entry(id: id), :put, '').
+          returns({"base" => {'id' => id, 'errors' => []}})
+      end
 
-        subject.save
+      it 'is expected to return true' do
+        expect(subject.update).to eq(true)
       end
     end
   end
 
-  describe 'update' do
-    let!(:id) { 42 }
+  describe 'create' do
+    before :each do
+      subject.expects(:save)
+      KalibroClient::Entities::Base.
+        expects(:new).
+        with({}).
+        returns(subject)
+    end
 
-    context 'with valid parameters' do
+    it 'should instantiate and save the model' do
+      expect(KalibroClient::Entities::Base.create {}).to eq(subject)
+    end
+  end
+
+  describe 'find' do
+    context 'with an inexistent id' do
       before :each do
-        KalibroClient::Entities::Base.
-              expects(:request).
-              with(':id', {base: {}, id: id}, :put, '').returns({"base" => {'id' => id, 'kalibro_errors' => []}})
-        subject.expects(:id).returns(id)
-        subject.expects(:to_hash).returns({})
+        # We should remove this call in the future: there's no need to call exists before a find.
+        subject.class.stubs(:exists?).with(0).returns(false)
+
+        subject.class.stubs(:request).with(':id', has_entry(id: 0), :get).
+          raises(KalibroClient::Errors::RecordNotFound)
       end
 
-      it 'is expect to return true' do
-        expect(subject.update).to be_truthy
+      it 'should raise a RecordNotFound error' do
+        expect { subject.class.find(0) }.to raise_error(KalibroClient::Errors::RecordNotFound)
       end
     end
 
-    context 'with invalid parameters' do
+    context 'with an existent id' do
       before :each do
-        KalibroClient::Entities::Base.
-              expects(:request).
-              with(':id', {base: {}, id: id}, :put, '').returns({"errors" => ["Error"]})
-        subject.expects(:id).returns(id)
-        subject.expects(:to_hash).returns({})
+        subject.class.stubs(:exists?).with(42).returns(true)
+        subject.class.expects(:request).with(':id', has_entry(id: 42), :get).
+          returns("base" => {'id' => 42})
       end
 
-      it 'is expect to return false' do
-        expect(subject.update).to be_falsey
+      it 'should return an empty model' do
```

Comment:
Using `should` for spec descriptions has been deprecated. Instead, `is expected to` fits better to the code.